app/code/AstoundDRudenko/Badge/Model/Attribute/Badge/Source.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+namespace AstoundDRudenko\Badge\Model\Attribute\Badge;
+
+use \Magento\Eav\Model\Entity\Attribute\Source\AbstractSource;
+
+/**
+ * Class Source
+ * @package AstoundDRudenko\Badge\Model\Attribute\Badge
+ */
+class Source extends AbstractSource
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function getAllOptions()
+    {
+        if (!$this->_options) {
```

Comment:
what if a new option would be set to an attribute in admin panel manually?
i guess you need to load all options from attribute, and not from a constant